packages/pluggableWidgets/video-player-native/src/VideoPlayer.tsx

```diff
@@ -36,27 +53,133 @@ export function VideoPlayer(props: VideoPlayerProps<VideoStyle>): ReactElement {
-                    setStatus(StatusEnum.READY);
-                    setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
-                }}
-                onError={() => setStatus(StatusEnum.ERROR)}
-                style={status !== StatusEnum.READY ? { height: 0 } : styles.video}
-                useTextureView={false}
-                resizeMode={props.aspectRatio ? "contain" : "stretch"}
-            />
-        </View>
+        <Fragment>
+            {isAndroid && (
+                <Modal
+                    isVisible={fullScreen}
+                    style={styles.fullScreenVideoPlayer}
+                    onBackButtonPress={() => {
+                        fullScreenHandler(false);
+                    }}
+                >
+                    <TouchableWithoutFeedback onPress={() => handleShowControls()}>
+                        <Video
+                            {...videoProps}
+                            ref={fullScreenPlayerRef}
+                            paused={false}
+                            onLoad={data => {
+                                setStatus(StatusEnum.READY);
+                                setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
+                                fullScreenPlayerRef.current?.seek(currentPlayTime);
+                            }}
+                            style={styles.fullScreenVideoStyle}
+                        />
+                    </TouchableWithoutFeedback>
+                    {showControls && (
+                        <TouchableOpacity
+                            style={styles.controlBtnContainerStyle}
+                            onPress={() => {
+                                fullScreenHandler(false);
+                            }}
+                        >
+                            <Icon name="fullscreen-exit" color="white" size={22} />
+                        </TouchableOpacity>
+                    )}
+                    {status === StatusEnum.LOADING && (
+                        <ActivityIndicator
+                            color={styles.indicator.color}
+                            size="large"
+                            style={styles.fullScreenActivityIndicatorStyle}
+                        />
+                    )}
+                    {status === StatusEnum.ERROR && (
+                        <Text style={styles.errorMessage}>The video failed to load :(</Text>
+                    )}
+                </Modal>
+            )}
+            <View style={styles.container}>
+                {status === StatusEnum.LOADING && <ActivityIndicator color={styles.indicator.color} size="large" />}
+                {status === StatusEnum.ERROR && <Text style={styles.errorMessage}>The video failed to load :(</Text>}
+                {!fullScreen && (
+                    <TouchableWithoutFeedback style={styles.container} onPress={() => handleShowControls()}>
+                        <Video
```

Comment:
This create element of `Video` seems similar to https://github.com/mendix/widgets-resources/pull/1303/files#diff-5b19cf908530b18852147d085549427ea128e2eb1da3dddad52826adb3d6aae7R117

Are you able to reduce code duplication here? e.g. create a function that returns `Video` with different props, based on what you pass in as arguments

Reply:
I think it will have a lot of branching and many conditions.

Reply:
Okay